editor/plugins/org.fusesource.ide.camel.editor/src/org/fusesource/ide/camel/editor/restconfiguration/RestConfigEditor.java

```diff
@@ -532,17 +530,19 @@ private void refreshRestSection() {
 		if (!ctx.getRestElements().isEmpty()) {
 			restList.setInput(ctx.getRestElements().values());
 			restList.setSelection(new StructuredSelection(restList.getElementAt(0)), true);
+		} else {
+			clearUI();
 		}
 	}	
 	
 	public void reload() {
 		ctx = getCamelContext(parentEditor.getDesignEditor().getModel());
 		refreshRestConfigurationSection();
-		clearUI();
 		refreshRestSection();
 		form.layout(true);
 		toolkit.decorateFormHeading(form.getForm());
-		setSelection(StructuredSelection.EMPTY);
+		selection = null;
+		setSelection(new StructuredSelection());
```

Comment:
why using new StructuredSelection() instead of StructuredSelection.EMPTY?
it was avoiding an object creation